Extensions/timestamps.js

```diff
@@ -68,15 +68,54 @@ XKit.extensions.timestamps = new Object({
 					XKit.storage.set("timestamps", `extension__setting__${x}`, this.preferences[x].value.toString());
 				}
 			}
+			XKit.storage.set("timestamps", "preference_conversion", "done");
+		}
+
+	},
+
+	convert_preferences: function() {
+		const conversion = {
+			only_relative: {show_absolute: false},
+			only_inbox: {inbox: true, posts: false},
+			do_reblogs: {op: true, reblogs: true},
+			only_original: {reblogs: false}
+		};
+		const order = [
+			"only_relative",
+			"only_inbox",
+			"do_reblogs",
+			"only_original"
+		];
```

Comment:
I *think* this can be a little more clear if It's formatted as follows:

```js
const new_preferences = Object.assign({}, *[
  /* ... */
].filter(([preference, _]) => 
    XKit.storage.get("timestamps", `extension__setting__${key}`, "") === 'true'
).map(([_, changes]) => changes));

Object.entries(new_preferences).forEach(([key, value]) => { /* set */ });
```

instead of it being an iteration over mutable state

but YMMV!